networking/src/certificate.rs

```diff
@@ -0,0 +1,30 @@
+//! Creating [`Certificate`]s.
+
+/// A public Certificate. You can distribute it freely to peers.
+#[derive(Clone, Debug)]
+pub struct Certificate(pub(crate) Vec<u8>);
```

Comment:
Both `Certificate` and `PrivateKey` either need to make their contents publicly visible or accessible via an `as_bytes()` type accessor. Right now I can't save the generated certificate across multiple runs.

Reply:
Fixed.